src/core/core.js

```diff
@@ -546,6 +548,25 @@ export default class Core {
     return queryTrigger;
   }
 
+  /**
+   * Determines whether or not a new url state should be pushed for a search.
+   *
+   * If queryTrigger is INITIALIZE, don't push an extra state on page load.
+   * If queryTrigger is QUERY_PARAMETER, this only occurs either on page load
+   * or when hitting the history back and forward buttons. For both cases we
+   * don't push an extra state.
+   * If queryTrigger is SUGGEST, this only occurs when clicking a spellcheck
+   * link, which reloads the page, so don't push an extra state.
+   *
+   * @param {QueryTriggers} queryTrigger SDK query trigger
+   * @returns {boolean}
+   */
+  shouldPushState (queryTrigger) {
```

Comment:
LG, I just want to make sure I understand this. Previously, we would always set the persistentStorage for things that were waiting in global storage (`context`, `referrerPageUrl`). Here, because we have a new push state to history paradigm (all or nothing), AND because we need context and referrerPageUrl to show up, we have to be picky about when we push a state. 

1. We don't push history on INITIALIZE otherwise we would never be able to navigate backwards in history on page load.
2. We don't push history on QUERY_PARAMETER we don't push for the same reason.
3. This case is very similar to the QUERY_PARAMETER case, it's just a special subcase

Reply:
yup! 3 is a subcase of 2.